src/instructlab/eval/mt_bench.py

```diff
@@ -20,34 +24,71 @@
 logger = setup_logger(__name__)
 
 
-class MTBenchEvaluator(Evaluator):
+class AbstractMTBenchEvaluator(Evaluator):
     """
-    Child class of an Evaluator for Multi-turn Benchmark (MT-Bench)
+    Abstract class of an MTBenchEvaluator for Multi-turn Benchmark (MT-Bench)
 
     Attributes
         model_name                  Name of the model to evaluate
         judge_model_name            Name of the judge model
         output_dir                  The directory to use for evaluation output
-        max_workers                 Max parallel workers to run the evaluation with
+        max_workers                 Max parallel workers to run the evaluation with (int or "auto")
+        serving_gpus                Number of gpus allocated for serving.  Used to tune with max_workers=auto.
         merge_system_user_message   Boolean indicating whether to merge system and user messages (required for Mistral based judges)
     """
 
-    name = "mt_bench"
-
     def __init__(
         self,
         model_name: str,
         judge_model_name: str,
         output_dir: str = "eval_output",
-        max_workers: int = 40,
+        max_workers: int | str = "auto",
+        serving_gpus: int | None = None,
         merge_system_user_message: bool = False,
     ) -> None:
         self.model_name = model_name
         self.judge_model_name = judge_model_name
         self.output_dir = output_dir
-        self.max_workers = max_workers
+        self.serving_gpus = serving_gpus
         self.merge_system_user_message = merge_system_user_message
 
+        if max_workers == "auto":
+            if serving_gpus is not None:
+                # Tune max_workers based on hardware configuration: min(#GPUs being used * 10, #CPU cores)
+                self.max_workers = min(
+                    max(serving_gpus, 1) * 10, multiprocessing.cpu_count()
+                )
+                logger.debug("Auto tuning max_workers to %s", self.max_workers)
+            else:
+                # Don't be too aggressive when serving_gpus isn't specified. Use half the cpu count.
+                self.max_workers = multiprocessing.cpu_count() // 2
+                logger.debug(
+                    "max_workers set to auto but serving_gpus is not specified. Defaulting to (cpu count / 2): %s",
+                    self.max_workers,
+                )
+        else:
+            if isinstance(max_workers, int):
+                logger.debug("max_workers specified as: %s", max_workers)
+                self.max_workers = max_workers
```

Comment:
should we check to make sure `max_workers` > 0 here?

Reply:
The check wasn't there before, but technically yes.  Will address.